docs/tls/CONFIGURATION.md

```diff
@@ -0,0 +1,164 @@
+2. Client Certificate Secret 
+
+#### Creating the CA Secret
+
+This Secret is required for either TLS / mTLS mode. It contains the CA certificate, and is used by NLK to validate the server certificate. 
+
+The easiest way to create the secret is by using `kubectl` (note: these examples assume the commands found in the [Self-signed certificates guide](docs/tls/CERTIFICATES.md) were run in `/tmp`):
+
+```bash
+kubectl create secret tls nlk-tls-ca-secret --cert=/tmp/ca.crt --key=/tmp/ca.key
+```
+
+#### Creating the Client Certificate Secret
+
+This Secret is only required for mutual TLS. It contains the client certificate, and is used by NLK to authenticate to NGINX Plus.
+
+The easiest way to create the secret is by using `kubectl`:
+
+```bash
+kubectl create secret tls nlk-tls-client-secret --cert=/tmp/client.crt --key=/tmp/client.key
+```
+
+#### Secrets as yaml files
+
+The Secrets can also be created using yaml files. The following is an example of a yaml file for the Client Secret:
+
+```yaml
+apiVersion: v1
+data:
+  tls.crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUVCVENDQXUyZ0F3SUJBZ0lVYkNkTnhGSllXc0FsdkEvMnNvMVJoQzQybmpZd0RRWUpLb1pJaHZjTkFRRUwKQlFBd1hURUxNQWtHQTFVRUJoTUNWVk14RGpBTUJnTlZCQWdNQlZOdmJHbGtNUkV3RHdZRFZRUUhEQWhNYjJOaApkR2x2YmpFVk1CTUdBMVVFQ2d3TVQzSm5ZVzVwZW1GMGFXOXVNUlF3RWdZRFZRUUxEQXRFWlhabGJHOXdiV1Z1CmREQWVGdzB5TXpBNU1ERXhOek15TURCYUZ3MHlOREE0TXpFeE56TXlNREJhTUYweEN6QUpCZ05WQkFZVEFsVlQKTVE0d0RBWURWUVFJREFWVGIyeHBaREVSTUE4R0ExVUVCd3dJVEc5allYUnBiMjR4RlRBVEJnTlZCQW9NREU5eQpaMkZ1YVhwaGRHbHZiakVVTUJJR0ExVUVDd3dMUkdWMlpXeHZjRzFsYm5Rd2dnRWlNQTBHQ1NxR1NJYjNEUUVCCkFRVUFBNElCRHdBd2dnRUtBb0lCQVFERVZiUUtzRVpCOWk2eTloYzNHOXV4cWVJME5iNEduU2FnS1BJUGZNQTEKQnkwQ0s3MStEdDNuaENFY3BNdFpuNUYzYnN0aGlTRk5rdWw3Q3Y1aTJGMDN0enhoUWw1Y3BHVlhDSUxqZk4yego4TjBtY29jTUs1ZWhicFViUHlwV0dkUDBmZmJaS1hPZWRKUHRNYlJId09YWmlpYXgrYWRVN0hpWkNNM2V0cWhGCmdnTHhqMWJXNlFOQjhQQitVM3FCYkRETExrQTYyQXdRMG1CR2xibzhtUVlXY0tXWGkwNkZ2amJkUGR0V2Q2RG4KZHRLN0RjR1pjOWUzT3ZJbTBDeXZqY1NGTFpUeWRhY1FvZjMxRlNYRk5xdHpIeis2ZUxWTHp6QTg3QU1MMXhNNApyTWRxSEJMTEF2cUVSVnJWZGNwdzFSVnlmak91NUsxUVMwK0RKSlFLaXNGNUFnTUJBQUdqZ2J3d2dia3dFd1lEClZSMGxCQXd3Q2dZSUt3WUJCUVVIQXdJd0hRWURWUjBPQkJZRUZEMmhvMEp2Vm41SEU5Mk0vR0tDdm5xUkFrejQKTUlHQ0JnTlZIU01FZXpCNW9XR2tYekJkTVFzd0NRWURWUVFHRXdKVlV6RU9NQXdHQTFVRUNBd0ZVMjlzYVdReApFVEFQQmdOVkJBY01DRXh2WTJGMGFXOXVNUlV3RXdZRFZRUUtEQXhQY21kaGJtbDZZWFJwYjI0eEZEQVNCZ05WCkJBc01DMFJsZG1Wc2IzQnRaVzUwZ2hRcWY3QnhGYTFDaHpwYUhERGdBL2V3UzRYUkV6QU5CZ2txaGtpRzl3MEIKQVFzRkFBT0NBUUVBQ2gxbUpDZy9WaUNWZUVSRzRjZ01yUUZjbE9YNVptUDZWNGl1S0pwK28zSU9TVzhQbjFHTwpRNTF6ekVtNG9JaTZOZTgrTlJ2cGd4cWVVK3M3Rk1qSVdYQ1l1RFV1ZmREalAwWEtsWGR4djJaVitoOHlnMUx4CnZLSFpBcHQrTTVzbE4rdVRDZmtPQ01BRkhaMEVFYzhoT1VlcGpiOWt5b01mQU1CMk9nNTMrb3grUzJxS2ZzMlEKOTJuWVpKRGttblg3aEl2VEFaN2dlVUVBY0djNkZudDNoWlpRYmVKbm1MTFl4ZUd2YzFzMmdNZFAyWnM5VDlpTwptSmFFQ283L3hqcmJFbzlmMHlNb3B1Z1hPcUVnVXQ5dmw1ekxHUHZTemFyTC9UVjFsMElHVVVQU3pFZXRsdzZxCjJIM2c5YlBPRHJpaUJPNFNHcXdPWVlhUW8vWlY3Mkg1Wmc9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
```

Comment:
I hope these are not real.  We should not include working cert artifacts in docs.

LS0tLS1CRUdJTiBDRVJUSUZJQ<--snip-->BDRVJUSUZJQ0FURS0tLS0tCg==
For example

Reply:
These are just certs that I generated to ensure the steps are correct. They have garbage DN data and use example.com for any hostnames.

Reply:
They should still be snipped for doc brevity.